### PR TITLE
For loops in array.rs

### PR DESCRIPTION
**Pull request by corincerami**

### CODE REVIEW

File: src/koans/array.rs [hunk 1]
```diff
@@ -88,3 +88,25 @@ fn complex_array_filter_map() {
   assert!(iterator.next().unwrap() == __);
   assert!(iterator.next().is_none());
 }
+
+// Arrays can also be iterated through using a for loop
+#[test]
+fn for_loops() {
+  let arr: [u64; 3] = [1, 2, 3];
+  let mut y: u64 = 1;
+  for x in arr.iter() {
```
Comment (crazymykl) [inline src/koans/array.rs]: Does `&arr` work here?

Comment (corincerami) [inline src/koans/array.rs]: As opposed to `arr.iter()`?

Comment (crazymykl) [inline src/koans/array.rs]: Yes, an array slice should implement the right traits, IIRC.

Comment (corincerami) [inline src/koans/array.rs]: I'll try it out. :+1: :rainbow:

**Comment by corincerami** [comment]
@crazymykl updated

**Pull request merged**